src/ocean/OceanAgreementsConditions.ts

```
@@ -35,19 +35,24 @@ export class OceanAgreementsConditions extends Instantiable {
             escrowReward
         } = this.ocean.keeper.conditions
 
-        await this.ocean.keeper.token.approve(
-            lockRewardCondition.getAddress(),
-            amount,
-            from.getId()
-        )
+        try {
+            await this.ocean.keeper.token.approve(
+                lockRewardCondition.getAddress(),
+                amount,
+                from.getId()
+            )
 
-        const receipt = await lockRewardCondition.fulfill(
-            agreementId,
-            escrowReward.getAddress(),
-            amount,
-            from && from.getId()
-        )
-        return !!receipt.events.Fulfilled
+            const receipt = await lockRewardCondition.fulfill(
+                agreementId,
+                escrowReward.getAddress(),
+                amount,
+                from && from.getId()
+            )
+
+            return !!receipt.events.Fulfilled
+        } catch {
+            return false
```

should we silence the error here? or rethrow?

---

The error is not good in general, is just a revert without explanation. Is more useful the logs if the verbose mode is enabled, and it will be shown in both cases.

---

ok, I guess proper error handling is then for another PR?